### ISSUE TITLE
KIA1107 Subset not found - but it actually works

### ISSUE THREAD

**Issue by FWiesner** [kiali/kiali#2918]
I've followed the instructions on how to get TLS origination working. (see https://istio.io/latest/docs/tasks/traffic-management/egress/egress-tls-origination/). I'm using mostly the same configuration as in the example, with few differences:
- I use a different URL to connect to
- My subset is named `tls-origination-0` instead of `tls-origination`
- My `DestinationRule`, `VirtualService` and `ServiceEntry` all have `.spec.exportTo` defined for local namespace exposure only.

When running `istioctl analyze -n <namespace>`, I get back
```✔ No validation issues found when analyzing namespace:```
<img width="796" alt="Bildschirmfoto 2020-06-18 um 21 26 26" src="https://user-images.githubusercontent.com/1433733/85063432-6a0cc580-b1aa-11ea-9cc2-ad9b653a6a0f.png">

So, Istio tooling doesn't see any configuration issue with my setup and the test requests also behave as expected. BUT, Kiali keeps reporting `KIA1107 Subset not found`, marking my `VirtualService` with the yellow warning icon. When I get to the YAML view, the whole `destination` underneath my `route` section of the virtual service is marked 
<img width="485" alt="Bildschirmfoto 2020-06-18 um 21 18 18" src="https://user-images.githubusercontent.com/1433733/85063188-113d2d00-b1aa-11ea-9fd5-5035051c3a56.png">

**Versions used**
Kiali: 1.18.2
Istio: 1.6.2
Kubernetes flavour and version: vanilla 1.18.3

**To Reproduce**
1. Follow the Istio TLS Origination tutorial (under Egress)
2. Open Kiali
3. The namespace is marked with an Istio Config Validation warning marker <img width="537" alt="Bildschirmfoto 2020-06-18 um 21 22 57" src="https://user-images.githubusercontent.com/1433733/85063112-f1a60480-b1a9-11ea-927d-923cd9e916f0.png">
4. Drill down

**Expected behavior**
As the configuration seems to be fine and works, there should be no warning marker

**Comment by jshaughn** [comment]
cc @xeviknal

**Comment by lucasponce** [comment]
@FWiesner thanks, would you mind to attach/paste your VirtualService, DestinationRule, Gateway resources ?

**Comment by xeviknal** [comment]
As @lucasponce suggested, it would be interesting to see those rules. The most important ones are the DestinationRules in this scenario. In those objects are the subsets defined. In case you don't have any subset defined in any DR, you should see the validation warning.

It is possible that this is working because the is only one subset defined, so there is no room for random behavior. But again, we need more details to have (in)validate hypothesis. 
Thanks.

**Comment by FWiesner** [comment]
Hi guys, sorry for the delay:
```yaml
apiVersion: security.istio.io/v1beta1
kind: PeerAuthentication
metadata:
  name: "default"
  namespace: istio-trials
spec:
  mtls:
    mode: PERMISSIVE
---
apiVersion: networking.istio.io/v1beta1
kind: ServiceEntry
metadata:
  name: orahub
  namespace: istio-trials
spec:
  hosts:
    - orahub.oci.oraclecorp.com
  ports:
    - number: 80
      name: http-0
      protocol: HTTP
    - number: 443
      name: tls-0
      protocol: TLS
  resolution: DNS
  location: MESH_EXTERNAL
---
apiVersion: networking.istio.io/v1beta1
kind: Gateway
metadata:
  name: my-egress-gw
  namespace: istio-trials
spec:
  selector:
    istio: egressgateway
  servers:
    - hosts:
        - orahub.oci.oraclecorp.com
      port:
        number: 80
        name: http-0
        protocol: HTTP
    - hosts:
        - orahub.oci.oraclecorp.com
      port:
        number: 443
        name: tls-0
        protocol: TLS
      tls:
        mode: PASSTHROUGH
        minProtocolVersion: TLSV1_2
---
apiVersion: networking.istio.io/v1beta1
kind: ServiceEntry
metadata:
  name: egress-gw
  namespace: istio-trials
spec:
  hosts:
    - istio-egressgateway.istio-system.svc.cluster.local
  ports:
    - number: 80
      name: http-0
      protocol: HTTP
    - number: 443
      name: tls-0
      protocol: TLS
  resolution: DNS
  location: MESH_INTERNAL
---
apiVersion: networking.istio.io/v1beta1
kind: DestinationRule
metadata:
  name: egressgateway-for-orahub
  namespace: istio-trials
spec:
  host: istio-egressgateway.istio-system.svc.cluster.local
  trafficPolicy:
    loadBalancer:
      simple: ROUND_ROBIN
---
apiVersion: networking.istio.io/v1alpha3
kind: DestinationRule
metadata:
  name: originate-tls-for-orahub
  namespace: istio-trials
spec:
  host: orahub.oci.oraclecorp.com
  subsets:
    - name: tls-origination
      trafficPolicy:
        loadBalancer:
          simple: ROUND_ROBIN
        portLevelSettings:
          - port:
              number: 443
            tls:
              sni: orahub.oci.oraclecorp.com
              mode: SIMPLE
    - name: tls-passthrough
---
apiVersion: networking.istio.io/v1beta1
kind: VirtualService
metadata:
  name: orahub-vs
  namespace: istio-trials
spec:
  hosts:
    - orahub.oci.oraclecorp.com
  gateways:
    - my-egress-gw
    - mesh
  http:
    - match:
        - gateways:
            - mesh
          port: 80
      route:
        - destination:
            host: istio-egressgateway.istio-system.svc.cluster.local
            port:
              number: 80
    - match:
        - gateways:
            - my-egress-gw
          port: 80
      route:
        - destination:
            subset: tls-origination
            host: orahub.oci.oraclecorp.com
            port:
              number: 443
  tls:
    - match:
        - gateways:
            - mesh
          port: 443
          sniHosts:
            - orahub.oci.oraclecorp.com
      route:
        - destination:
            host: istio-egressgateway.istio-system.svc.cluster.local
            port:
              number: 443
    - match:
        - gateways:
            - my-egress-gw
          port: 443
          sniHosts:
          - orahub.oci.oraclecorp.com
      route:
        - destination:
            subset: tls-passthrough
            host: orahub.oci.oraclecorp.com
            port:
              number: 443
```

**Comment by xeviknal** [comment]
@FWiesner Thanks replying. I see the problem. The fact that you are using ServiceEntries is making this validation appear. We'll fix that and ping you when this is released.
Again, thanks for opening the issue and helping us increasing the value.

**Comment by xeviknal** [comment]
@FWiesner hi there! We've put some pieces together to cover your scenario. thanks for sharing and contributing to have a better product. This fix will be available into the next release, kiali v1.20.

**Comment by FWiesner** [comment]
Awesome - thanks!
